content/guides/reader_conditionals.adoc

```diff
@@ -0,0 +1,198 @@
+
+[source,clojure]
+----
+(ns rethinkdb.query
+  (:require [clojure.walk :refer [postwalk postwalk-replace]]
+            [rethinkdb.query-builder :as qb :refer [term parse-term]]
+            #?(:clj [rethinkdb.net :as net])))
+
+;; snip...
+
+#?(:clj (defn run [query conn]
+      (let [token (get-token conn)]
+        (net/send-start-query conn token (replace-vars query)))))
+----
+
+== Exception handling
+
+Exception handling is another area that benefits from reader conditionals. ClojureScript supports `(catch :default)` to catch everything and this may come Clojure too, however you will often still want to handle host specific exceptions. Here's an https://github.com/runexec/lemon-disc/blob/c24c6638f1d476a0f5470387e52a2b702117c4a9/src/lemon_disc/core.cljc#L65-L72[example] from https://github.com/runexec/lemon-disc[lemon-disc].
+
+[source,clojure]
+----
+(defn message-container-test [f]
+  (fn [mc]
+      (passed?
+        (let [failed* (failed mc)]
+          (try
+            (let [x (:data mc)]
+              (if (f x) mc failed*))
+            (catch #?(:clj Exception :cljs js/Object) _ failed*))))))
+----
+
+== Splicing
+
+The splicing reader conditional is not as widely used as the standard one. For an example on its usage, let's get really meta and look at the https://github.com/clojure/clojure-clr/blob/544e9354e121e10a656702222d47c8398468fb02/Clojure/Clojure.Tests/clojure/test_clojure/reader.cljc#L672-L677[tests] for reader conditionals in the ClojureCLR reader. What might not be obvious at first glance is that the vectors inside the splicing reader conditional are being wrapped by a surrounding vector.
+
+[source,clojure]
+----
+(deftest reader-conditionals
+     ;; snip
+     (testing "splicing"
+              (is (= [] [#?@(:clj [])]))
+              (is (= [:a] [#?@(:clj [:a])]))
+              (is (= [:a :b] [#?@(:clj [:a :b])]))
+              (is (= [:a :b :c] [#?@(:clj [:a :b :c])]))
+              (is (= [:a :b :c] [#?@(:clj [:a :b :c])]))))
+----
+
+== File organisation
+
+There isn't a clear community consensus yet around where to put `.cljc` files. Two options are to have a `src` directory where `.clj`, `.cljs`, and `.cljc` files, or to have separate `src/clj`, `src/cljc`, and `src/cljs` directories.
+
+== CLJX
+
+Before Reader Conditionals, the same goal of sharing code between platforms was solved by https://github.com/lynaghk/cljx[cljx]. cljx processes Clojure files with a .cljx extension and output multiple platform specific files to a generated sources directory. These were then read as normal Clojure or ClojureScript files by the Clojure <<xref/../../reference/reader#,reader>>. This worked well, but required another piece of tooling to run, and it wasn't able to be used in Clojure contrib projects. cljx was deprecated on June 13 2015 in favour of reader conditionals.
```

Comment:
- "Before reader conditionals were introduced..." "a Leiningen plugin called cljx"
- "processes Clojure files" => "processes files with the extension `.cljx` and outputs"
- " and it wasn't able to be used in Clojure contrib projects" -- drop this? Seems not particularly relevant